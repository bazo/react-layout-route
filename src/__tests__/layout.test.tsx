import React from "react";
import WithLayout, { LayoutProps } from "../";
import { render } from "enzyme";
import { MemoryRouter, RouteProps } from "react-router";

const Layout = ({ children, ...props }: { children: React.ReactNode } & LayoutProps<any>) => {
	return (
		<div>
			<h1>My Layout</h1>
			{children}
		</div>
	);
};

const Child = (props: LayoutProps<{ id: number }>) => {
	return <div>child {props.match.params.id}</div>;
};

describe("with layout", () => {
	test("renders layout with children", () => {
		const MyLayout = (props: RouteProps) => <WithLayout {...props} layout={Layout} />;

		const wrapper = render(
			<MemoryRouter initialEntries={["/test/1"]}>
				<MyLayout path="/test/:id" component={Child} />
			</MemoryRouter>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
