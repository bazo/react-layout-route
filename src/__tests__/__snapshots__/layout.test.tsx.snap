// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with layout renders layout with children 1`] = `
<div>
  <h1>
    My Layout
  </h1>
  <div>
    child 1
  </div>
</div>
`;
